### ISSUE TITLE
Missclassifying public field when using coma to initialize several at once in C#.

### ISSUE THREAD

**Issue by AleixFerre** [doxygen/doxygen#8718]
**Describe the bug**
I have two public int fields initialized in this class. Then I have several private ones. The program missunderstand the coma separated field and place it into the private section.

**Expected behavior**
Obviously the second int would need to be public aswell.

**Screenshots**
![image](https://user-images.githubusercontent.com/12767963/129432518-667ca437-a1d4-4497-b4ea-da651d351fa9.png)

```cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

[RequireComponent(typeof(Collider), typeof(TurretStats))]
public class BaseRegenerator : MonoBehaviour {

    public int HPRegen, MPRegen; //< Here it happens!

    private List<PlayerStats> playersInside = new List<PlayerStats>();
    private float lastRegenTime;
    private Collider _collider;
    private TurretStats stats;
   
    // ...

```

**Version**
Windows 10 - 64bit
```bash
$ doxygen -v
1.9.1 (ef9b20ac7f8a8621fcfc299f8bd0b80422390f4b)
```

**Additional context**
This code is from a game I'm developing in Unity using C#.

**Comment by albert-github** [comment]
When setting:
```
EXTRACT_ALL = YES
EXTRACT_PRIVATE = YES
```
I can reproduce the problem.

To be sure that we speak about the same:
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by AleixFerre** [comment]
Sure, this is my output for `./proj/html/class_example.html`

![image](https://user-images.githubusercontent.com/12767963/129450850-49c78464-d551-49c7-99be-430fa175681f.png)

The Doxyfile is the autogenerated with `doxygen -g Doxyfile`, the only think I changed is 
```
EXTRACT_ALL = YES
EXTRACT_PRIVATE = YES
```

All the code is (the only file i have is `./proj/test.cs`)
```cs
public class Example {
    public int a, b;
    private float c;
}
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8720.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.2.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
